lib/deno/mod.ts

```diff
@@ -209,6 +211,11 @@ const spawnNew: SpawnFn = (cmd, { args, stdin, stdout, stderr }) => {
     write: bytes => writer.write(bytes),
     read: () => reader.read().then(x => x.value || null),
     close: async () => {
+      // Ref the child process again, so that a user calling `close()` can await
+      // the returned promise without the event loop starving because there are
+      // no more ref'd async tasks.
+      child.ref()
```

Comment:
This is the only new code. All other code is just a revert of 116f63ec1d063b5f75c4b3ba959da8c7598c495b.